File: domain/dto/page.go
package dto

// Page is a data transfer object for the page domain.
type Page struct {
	ID string `json:"id"`
	Title string `json:"title"`
	Description string `json:"description"`
	Content *string `json:"content"`
	IsBlog bool `json:"isBlog"`
	IsActive bool `json:"isActive"`
	ImageID *string `json:"imageId"`
	URL string `json:"url"`

	Audit []*PageAudit `json:"audit,omitempty"`
	SEO *SEO `json:"seo,omitempty"`
}

File: admin-client/src/app/models/page.ts
import Seo from "./seo";

interface CreatePage {
    title: string;
    description: string;
    content: string;
    url: string;
    seo: Seo;
}

interface UpdatePage {
    id: string;
    title: string;
    description: string;
    content: string;
    url: string;
    seo?: Seo;
}

interface Page {
    id: string;
    title: string;
    description: string;
    content: string;
    isBlog: boolean;
    isActive: boolean;
    imageId?: string;
    url: string;
    audit?: PageAudit[];
    seo?: Seo;
}

interface PageAudit {
    UserFullname: string;
    UserId: string;
    Date: Date;
    Message: string;
}

export { CreatePage, UpdatePage, Page, PageAudit };
